src/backend/distributed/utils/multi_partitioning_utils.c

```diff
@@ -569,7 +569,18 @@ CreateFixPartitionShardIndexNamesTaskList(Oid parentRelationId, Oid partitionRel
 			task->taskId = taskId++;
 
 			task->taskType = DDL_TASK;
-			SetTaskQueryStringList(task, queryStringList);
+
+			/*
+			 * In order to avoid round-trips per query in queryStringList, we
+			 * join the string and send as a single command.
+			 */
+			char *string = StringJoin(queryStringList, ';');
+			StringInfo commandToRun = makeStringInfo();
+
+			appendStringInfo(commandToRun,
+							 "SELECT pg_catalog.citus_run_local_command($$%s$$)", string);
+			SetTaskQueryString(task, commandToRun->data);
```

Comment:
We previously were able to concatenate queries and put them into a single task. Is this because of binary encoding changes? (using SendRemoteCommandParams instead of SendRemoteCommand)

Reply:
I guess I should have clarified a little further on this.

To start with `queryStringList` can be really long, because it may contain #partitions * #indexes entries.

Before this change, we were actually going through the execution you mentioned via `SetTaskQueryStringList`. However, that code-path triggers 1 round trip per entry in `queryStringList`.

The aim of this patch is to avoid the round-trips by creating a single query string.

I first simply tried sending `q1;q2;..;qn` . However, the executor is designed to handle `q1;q2;..;qn` type of query executions via the infrastructure you mentioned (e.g., by tracking the query indexes in the list). 

One another option could have been to change the executor such that only track the query index when `queryStringList` is provided not with `queryString` including multiple `;`s .  That is (a) more work (b) could cause weird edge cases with failure handling (c) felt like coding a special case in to the executor 

Hence, I decided to propose the trick here.

If you are fine with the above, I can try to document this in the code a little further,

Reply:
Batching is a more general problem, which will require a more general solution (e.g. can SendNextQuery concatenate stuff?). Anyway, everyone always ends up creating a UDF that does EXECUTE, so not much harm in adding one to Citus.